lib/bump/dependency_file_parsers/python.rb

```diff
@@ -14,9 +13,59 @@ def initialize(dependency_files:)
       end
 
       def parse
-        PythonHelpers.
-          parse_requirements(@requirements.content).
-          map { |dep| Dependency.new(name: dep[0], version: dep[1]) }
+        @requirements.
+          content.
+          each_line.
+          each_with_object([]) do |line, dependencies|
+            dependency = LineParser.new(line).parse
+
+            next if dependency.nil?
+            next if dependency[:requirements].length.zero?
+
+            # Ignore dependencies with multiple requirements, since they would
+            # cause trouble at the dependency update step
+            next if dependency[:requirements].length > 1
+
+            version =
+              dependency[:requirements].first.match(LineParser::VERSION)[0]
+
+            dependencies << Dependency.new(
+              name: dependency[:name],
+              version: version,
+              language: "python"
+            )
+          end
+      end
+
+      class LineParser
+        NAME = /[a-zA-Z0-9\-_\.]+/
+        EXTRA = /[a-zA-Z0-9\-_\.]+/
+        COMPARISON = /===|==|>=|<=|<|>|~=|!=/
+        VERSION = /[a-zA-Z0-9\-_\.]+/
+        REQUIREMENT = /#{COMPARISON}\s*#{VERSION}/
+
+        REQUIREMENT_LINE =
+          /^\s*(?<name>#{NAME})
+            \s*(\[\s*(?<extras>#{EXTRA}(\s*,\s*#{EXTRA})*)\s*\])?
+            \s*(?<requirements>#{REQUIREMENT}(\s*,\s*#{REQUIREMENT})*)?
+            \s*#*\s*(?<comment>.+)?$
+          /x
+
+        attr_reader :line
+
+        def initialize(line)
+          @line = line
+        end
+
+        def parse
+          requirement = line.chomp.match(REQUIREMENT_LINE)
+          return if requirement.nil?
+
+          {
+            name: requirement[:name],
+            requirements: (requirement[:requirements] || "").scan(REQUIREMENT)
```

Comment:
It would be cool if `LineParser` would return the comparison and the version separately, so we don't have to do any more matching outside of it. Something like
```ruby
{
  name: "luigi",
  requirements: [
    { version: "1.0.0", comparison: ">=" },
    { version: "2.0.0", comparison: "<" }
  ]
}
```

or

```ruby
{
  name: "luigi",
  requirements: [
    [">=", "1.0.0"],
    ["<", "2.0.0"]
  ]
}